staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1597,6 +1597,30 @@ func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
 		return e.KeyFunc(genericapirequest.NewContext(), accessor.GetName())
 	}
 
+	// reverse key into namespace and name
+	// Examples:
+	//		key:    /registry/pods/namespace/name
+	//		prefix: /pods
+	reverseKeyFunc := func(key string) (name string, namespace string, err error) {
+		tokens := strings.Split(key, prefix)
+		if len(tokens) != 2 {
+			err = fmt.Errorf("invalid key %q, requiring resource prefix %q", key, prefix)
+			return
```

Comment:
can you please use explicit return with all the 3 values in all the empty `return`(s)? makes it easier to read